### ISSUE TITLE
PIDFILE contains only a single PID after starting multiple workers

### ISSUE THREAD

**Issue by Z0Marlin** [resque/resque#1721]
When starting multiple workers using rake task `resque:workers` and the `PIDFILE` env variable, only a single pid is available in the specified file, although multiple workers were started.

**Comment by tjozwik** [comment]
@Z0Marlin it is caused because in fact `resque:workers` calls `resque:worker` task `COUNTER` times. Besides, I think that there is a race condition for the `PIDFILE` access.

**Comment by Z0Marlin** [comment]
True. What could be a solution to this? I think appending to the file instead of overwriting and having a file lock should solve the problem. The file can be truncated initially to remove PIDs of the previous execution.

**Comment by tjozwik** [comment]
Yeah, that sounds great for me. Additionally, there will be a `resque:worker` task's refactor required as we would need to truncate the file for single-worker requests too. So, in general I think we can fix it in such a way:
1. Rename `resque:worker` task to e.g. `resque:run`,
2. Create `resque:worker` task that will be truncating `PIDFILE` (if env variable is defined, maybe using additional task for truncating) and calling `resque:run` once,
3. Change `resque:workers` task to truncate `PIDFILE` and use `resque:run` instead `resque:worker`,
4. Change `prepare` method in `lib/resque/worker.rb` so as to lock `PIDFILE` (looks like `File#flock`) and append current worker `PID` instead of overwriting it.

Unfortunately, I am afraid I don't have a time for creating PR for this issue right now/soon.

**Comment by Z0Marlin** [comment]
No problem. Thanks for the describing the changes required. I'll start working on the patch.
